### ISSUE TITLE
enable https / ssl authentication

### ISSUE THREAD

**Issue by jhamman** [pangeo-data/pangeo-binder#11]
Currently, pangeo-binder is a fairly insecure. I've tried adding the https config to the binderhub chart but it seems like there is a kubernetes api version mismatch: https://github.com/jupyterhub/zero-to-jupyterhub-k8s/issues/475

Recording what I tried here and we'll come back to this. 

```
  ingress:
    enabled: true
    annotations:
      kubernetes.io/ingress.class: nginx
    https:
      enabled: true
      type: "kube-lego"
  kube-lego:
    config:
      LEGO_EMAIL: $EMAIL
      LEGO_URL: https://acme-v01.api.letsencrypt.org/directory
    rbac:
      create: true
```

**Comment by mrocklin** [comment]
cc @dsludwig in case he has suggestions

**Comment by jhamman** [comment]
pinging @dsludwig who seems to understand the path forward here.

**Comment by guillaumeeb** [comment]
Si @dsludwig, did you try to deploy binder.pangeo.io with this setup?

**Comment by guillaumeeb** [comment]
Thanks @dsludwig, that is great news!

**Comment by jhamman** [comment]
closed via #19

**Comment by guillaumeeb** [comment]
:tada: Thanks @dsludwig and @jhamman!
